### PR TITLE
gateway: improvements to status writing

### PR DESCRIPTION
**Pull request by howardjohn**
This contains a few improvements to Gateway status, split out from https://github.com/istio/istio/pull/43889

* Support writing status for ambient GatewayClass, and fix some code de-duping there.
* Remove deprecated status messages and Ready (which we cannot implement). See https://github.com/kubernetes-sigs/gateway-api/issues/1832 for more info.
* Always report Gateway status from the main controller, even when we manually deploy it. This allows us to more accurately report status and simplifies things quite a bit. The main motivation is this allows us to properly report the address for waypoints, needed due to https://github.com/istio/istio/pull/43889.

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 1]
```diff
@@ -1422,11 +1370,16 @@ func convertGateways(r ConfigContext) ([]config.Config, map[parentKey][]*parentI
...
-			server, ok := buildListener(r, obj, l, i)
+			server, ok := buildListener(r, obj, l, i, controllerName)
 			if !ok {
 				invalidListeners = append(invalidListeners, l.Name)
 				continue
 			}
+			servers = append(servers, server)
+			if controllerName == constants.ManagedGatewayMeshController {
+				// Waypoint doesn't actually convert the routes to VirtualServices
+				continue
```
Comment (GregHanson) [inline pilot/pkg/config/kube/gateway/conversion.go:1381]: will waypoint ever support other fields like routes?  Wondering if istio needs to provide any sort of `unsupported fields` warnings for `istio-waypoint` gateways?

Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/conversion.go:1381]: It might, but not right now

**Review by GregHanson — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]
LGTM

**Review by ericvn — APPROVED** [review]

**Review by therealmitchconnors — COMMENTED** [review]
UX changes lgtm, but it seems to be missing integ tests.

**Comment by howardjohn** [comment]
> UX changes lgtm, but it seems to be missing integ tests.

The bulk of this is handled by unit tests which show the status writing output, and Gateway conformance tests which ensure we set things properly. However, there are some gaps mostly around ambient - will update those tests.

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**